### ISSUE TITLE
[OT Documentation Tables] Tables do not resize as wide as the web browser window

### ISSUE THREAD

**Issue by jonmichelson** [lowRISC/opentitan#19078]
### Description

Tables within the OT documentation web pages have a maximum width (regardless of the width of the web browser window). This makes viewing the tables within the programmers guide difficult.

For example, the tables linked below require a lot of vertical scrolling to move between the rows you are trying to read and the horizontal scroll bar at the bottom to change the viewable columns.

https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#error-handling
https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#direct-access-memory-map

Please allow the tables to expand all the way to the width of wide browser windows.

Thanks!

**Comment by hcallahan-lowrisc** [comment]
Thanks for raising this issue, I've been meaning to do something about this for a while!

Formatting of large tables is a bit tricky in general, in my experience most sites just avoid the problem by keeping the number of columns in any one table small. (if you have any examples of sites that present large tables well, please let me know!) Generally you want to [limit the line length](https://baymard.com/blog/line-length-readability) of your main page body for reader accessibility, and many sites choose to constrain themselves to just stick all page "content" into this bucket too.
It also helps to keep tables narrow when you consider publishing to different media, such as a pdf or other offline document that is normally limited to print-media widths. We are also trying to maintain readable docs when you view the source-code in other viewers, such as the github browser [[1]](https://github.com/lowRISC/opentitan/blob/43ad38b5e3e1fd197eed5d0d178ebb2a0e722d41/hw/ip/otp_ctrl/doc/programmers_guide.md) [[2]](https://github.com/lowRISC/opentitan/blob/43ad38b5e3e1fd197eed5d0d178ebb2a0e722d41/hw/ip/aes/doc/interfaces.md).

That's not to say we couldn't come up with an elegant way to present large tables on the docs site. In fact, maybe this is something we could choose to prioritize and, for example, publish the docs as an offline website to make the most of effort spent optimizing the user experience here.

If anyone else wants to weigh in here, please do. Longer-term plans may influence how we handle this issue. It may make more sense to refactor larger tables into smaller ones, or make tweaks to the current styling, such as making scrollbars always visible on larger tables no matter where you are currently viewing.

**Comment by jonmichelson** [comment]
Thanks for the background.

One extra complication to note is that the tables in question are software generated, making it harder for contributors to know exactly how wide the table will be when filling out the data elsewhere that ends up in the table.

The table in the "Error Handling section" rendered much more nicely in the github browser. But some tables aren't visible at all in github browser view. For example, a couple paragraphs under "[Direct Access Memory Map](https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#direct-access-memory-map)", there's a table that just shows as this in github browser:

{{#include otp_ctrl_mmap.md}}

Maybe that's a separate problem to be fixed?

If having scrollbars always visible is simple enough to implement, that's my vote for incremental improvement for now, although I'd really like in the longer run also to be able to expand the table width to the width of the browser window.

Thanks!

**Comment by hcallahan-lowrisc** [comment]
> The table in the "Error Handling section" rendered much more nicely in the github browser. But some tables aren't visible at all in github browser view. For example, a couple paragraphs under "[Direct Access Memory Map](https://opentitan.org/book/hw/ip/otp_ctrl/doc/programmers_guide.html#direct-access-memory-map)", there's a table that just shows as this in github browser:
> 
> {{#include otp_ctrl_mmap.md}}
> 
> Maybe that's a separate problem to be fixed?

Yep, those `{{ }}` blocks are replaced with generated content as part of building the site, which clearly goes against our goal of readable docs as source code. We have some new tooling ("CMDGEN") to address this, but it has not yet been rolled out across the entire repository. You can see the initial tooling PR in https://github.com/lowRISC/opentitan/pull/18235 and also https://github.com/lowRISC/opentitan/pull/18694 as the first changeset to use the tooling.

> If having scrollbars always visible is simple enough to implement, that's my vote for incremental improvement for now, although I'd really like in the longer run also to be able to expand the table width to the width of the browser window.

I'll see what I can do re-scrollbars to start with. Thanks :)

**Comment by rivos-eblot** [comment]
Another somewhat related weird bug w/ Safari:
![ot-bug](https://github.com/lowRISC/opentitan/assets/115395740/f7cff08d-cf1c-4cd7-995d-60a0b5212de4)

BTW, I've also notice that once the left menu bar is hidden, jumping to the previous/next page using the large arrow buttons at the bottom of the current page makes the left menu bar to re-appear. I think it should not till the icon that made it disappear is not pressed once more.
